core/src/main/java/org/elasticsearch/bootstrap/Security.java

```diff
@@ -212,7 +226,36 @@ static Permissions createPermissions(Environment environment) throws IOException
             // we just need permission to remove the file if its elsewhere.
             policy.add(new FilePermission(environment.pidFile().toString(), "delete"));
         }
-        return policy;
+    }
+    
+    static void addBindPermissions(Permissions policy, Settings settings) throws IOException {
+        // http is simple
+        String httpRange = settings.get("http.netty.port", 
+                               settings.get("http.port", 
+                                       NettyHttpServerTransport.DEFAULT_PORT_RANGE));
+        policy.add(new SocketPermission("localhost:" + httpRange, "listen,resolve"));
+        // transport is waaaay overengineered
+        Map<String, Settings> profiles = settings.getGroups("transport.profiles", true);
```

Comment:
> That is why instead I would just prefer to keep the security code contained and not "reach out" to other code. 

@rmuir Completely agree (I think I made a simultaneous edit to my [comment](https://github.com/elastic/elasticsearch/pull/14549#discussion_r44182190) to express this while you were writing the comment to which this is a reply).